### PR TITLE
[SD-6] Add week for periods

### PR DESCRIPTION
**Pull request by Zviad1990**

### CODE REVIEW

File: tap_dashboard_reports/period.py [hunk 1]
```diff
@@ -50,6 +51,28 @@ def _calculate_start_date(self):
         return self._end_date - relativedelta(days=self._periods)
 
 
+class PeriodWeek(PeriodDefault):
+    def iterate(self):
+        cursor = self._start_date
+        while cursor < self._end_date:
+            interval_start = cursor
```
Comment (alexkv) [inline tap_dashboard_reports/period.py]: Start date is not Monday here

File: tap_dashboard_reports/period.py [hunk 2]
```diff
@@ -50,6 +51,28 @@ def _calculate_start_date(self):
...
+            interval_start = (
+                self._start_date
+                if interval_start < self._start_date
+                else interval_start
+            )
+            interval_end = previous_day_of_week(cursor)
```
Comment (alexkv) [inline tap_dashboard_reports/period.py]: This name doesn't make sense, in should not be the previous day, it should be the last date of the week

File: tap_dashboard_reports/period.py [hunk 3]
```diff
@@ -78,3 +101,8 @@ def last_day_of_month(dt):
 
 def first_day_of_month(dt):
     return dt.replace(day=1)
+
+
+def previous_day_of_week(dt):
```
Comment (alexkv) [inline tap_dashboard_reports/period.py]: If `dt` is Monday the result of this function will be the next Monday, but should be Sunday, the last day of the week

**Review by alexkv — CHANGES_REQUESTED** [review]

File: tap_dashboard_reports/period.py (hunk 2 above)
Comment (Zviad1990) [reply tap_dashboard_reports/period.py]: changed

File: tap_dashboard_reports/period.py (hunk 3 above)
Comment (Zviad1990) [reply tap_dashboard_reports/period.py]: changed to 
```
def last_day_of_week(dt):
    return dt - timedelta(days=dt.weekday()) + relativedelta(weeks=1) - relativedelta(days=1)
```
always return sunday of the week

File: tap_dashboard_reports/period.py (hunk 1 above)
Comment (Zviad1990) [reply tap_dashboard_reports/period.py]: return 
```
def first_day_of_week(dt):
    return dt - timedelta(days=dt.weekday())
```

always returns mondays of the week

**Review by alexkv — APPROVED** [review]
LGTM 👍

**Pull request merged**